devito/finite_differences/coefficients.py

```diff
@@ -244,7 +244,7 @@ def generate_subs(deriv_order, function, dim):
     # Determine which 'rules' are missing
     sym = get_sym(functions)
     terms = obj.find(sym)
-    args_present = filter_ordered(term.args[1:] for term in terms)
+    args_present = filter_ordered([term.args[1:] for term in terms])
```

Comment:
I think we should just use `as_tuple` inside `filter_ordered` so that it's fine to pass in generators

Reply:
Probably yeah, on it